step03/migrations/versions/3b2e4aa45b69_user_orm_split_fullname_into_first_and_.py

```diff
@@ -0,0 +1,56 @@
+"""user ORM: added first and last
+
+Revision ID: 3b2e4aa45b69
+Revises: 337811a3f6b0
+Create Date: 2020-07-18 18:53:27.989200
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import table
+
+
+# revision identifiers, used by Alembic.
+revision = '3b2e4aa45b69'
+down_revision = '337811a3f6b0'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('users', sa.Column('first_name', sa.String(), nullable=True))
+    op.add_column('users', sa.Column('last_name', sa.String(), nullable=True))
```

Comment:
Be careful of avoiding weird intermediate states in schema/data migration. For example, if anything fails after migrating schema but before migrating data, it will be stuck in a state where you can neither upgrade (because the schema is already migrated) nor downgrade (because the data migration may have been destructive)

Reply:
Added a try block to help prevent this.